### PR TITLE
parser: grammar simplification

### PR DESCRIPTION
**Pull request by egabancho**
* INCOMPATIBLE Removes nestable keywords (to be re added in the future).

* INCOMPATIBLE Removes optional space after a keyword.

* INCOMPATIBLE `AND`, `OR` and `NOT` are only considered as keywords
  when written in caps.

### CODE REVIEW

**Comment by lnielsen** [comment]
Phrase query for double quotes

File: invenio_query_parser/utils.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def build_valid_keywords_grammar():
+    """."""
+    try:
+        import re
+        from flask import current_app
+        from pypeg2 import attr
+
+        from .parser import (
```
Comment (lnielsen) [inline invenio_query_parser/utils.py]: I think normally it's better not to do relative imports inside functions

File: invenio_query_parser/utils.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        KeywordRule.grammar = attr('value', re.compile(
+            r"(\d\d\d\w{0,3}|%s)\b" % "|".join(keywords_list), re.I))
+
+        NotKeywordValue.grammar = attr('value', re.compile(
+            r'\b(?!\d\d\d\w{0,3}|%s:)\S+\b:' % ":|".join(keywords_list)))
+
+        SimpleQuery.grammar = attr(
+            'op', [NotKeywordValue, KeywordQuery, ValueQuery])
+    except ImportError:
```
Comment (lnielsen) [inline invenio_query_parser/utils.py]: We've mostly been using [this pattern](https://www.quantifiedcode.com/knowledge-base/correctness/Catching%20ImportError%20to%20detect%20package%20availability/2LHXeS4F) for checking package availability these days.

File: invenio_query_parser/utils.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio-Query-Parser.
+# Copyright (C) 2014, 2015 CERN.
```
Comment (lnielsen) [inline invenio_query_parser/utils.py]: Year

File: invenio_query_parser/ast.py [hunk 4]
```diff
@@ -152,6 +152,10 @@ class KeywordOp(BinaryOp):
     pass
 
 
+class NestedKeywordsRule(BinaryOp):
```
Comment (lnielsen) [inline invenio_query_parser/ast.py]: Year

**Comment by lnielsen** [comment]
LGTM. Once ready, I think you can just remove all the commented code/fixme's.

**Comment by egabancho** [comment]
Do you think I should remove the tests? At some point they should work again, maybe ...

**Comment by lnielsen** [comment]
I think we can simply remove them as discussed IRL with @tiborsimko.

**Pull request merged**